### PR TITLE
add record types

### PR DESCRIPTION
**Pull request by sorja**
Part of #3

### CODE REVIEW

**Review by minotogna — CHANGES_REQUESTED** [review]
add Node in this PR as well

File: src/record/index.ts [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+  ownerUuid: string
+  step: string
+  dateCreated: Date
+  surveyUuid: string
+  dateModified: Date
+  ownerName: string
+  clusterId: string
+  measurement: string
+  surveyId: string
+  validation: Validation
```
Comment (minotogna) [inline src/record/index.ts:14]: can the properties be sorted alphabetically ?

File: src/record/index.ts [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+export interface Record {
+  uuid: string
+  ownerUuid: string
+  step: string
+  dateCreated: Date
+  surveyUuid: string
+  dateModified: Date
+  ownerName: string
+  clusterId: string
+  measurement: string
```
Comment (minotogna) [inline src/record/index.ts:12]: these two do not belong here

File: src/record/index.ts [hunk 3]
```diff
@@ -0,0 +1,15 @@
+import { Validation } from 'src/validation/index'
+
+export interface Record {
+  uuid: string
+  ownerUuid: string
+  step: string
+  dateCreated: Date
+  surveyUuid: string
+  dateModified: Date
```
Comment (minotogna) [inline src/record/index.ts:9]: this is a string

File: src/record/index.ts [hunk 4]
```diff
@@ -0,0 +1,15 @@
+import { Validation } from 'src/validation/index'
+
+export interface Record {
```
Comment (minotogna) [inline src/record/index.ts:3]: cycle is missing

Comment (minotogna) [inline src/record/index.ts:3]: preview is missing

File: src/record/index.ts [hunk 5]
```diff
@@ -0,0 +1,15 @@
...
+  uuid: string
+  ownerUuid: string
+  step: string
+  dateCreated: Date
```
Comment (minotogna) [inline src/record/index.ts:7]: this is a string

File: src/record/index.ts (hunk 4 above)
Comment (minotogna) [inline src/record/index.ts:3]: nodes is missing

File: src/node/index.ts [hunk 6]
```diff
@@ -0,0 +1,23 @@
...
+
+export interface Node {
+  dateCreated: string
+  dateModified: string
+  id: string
+  meta: NodeMeta
+  nodeDefUuid: string
+  parentUuid?: string
+  recordUuid: string
+  refData?: CategoryItem
```
Comment (minotogna) [inline src/node/index.ts:20]: this can also be a CategoryItem | Taxon

**Review by SteRiccio — APPROVED** [review]

**Review by minotogna — APPROVED** [review]

**Pull request merged**